### ISSUE TITLE
YARP bindings generation not working on devel branch

### ISSUE THREAD

**Issue by nunoguedelha** [robotology-playground/yarp-matlab-bindings#9]
I haven't tried on `master` branch, but for sure it's not working on `devel` branch. When generating the `yarpMATLAB_wrap.cxx` file, the "swig" keyword seems to be missing in the line command, so the next keyword `matlab` is interpreted as a command and Matlab is launched! refer to the following compilation sample:
```
$ xcodebuild -verbose -target ALL_BUILD
=== BUILD AGGREGATE TARGET ZERO_CHECK OF PROJECT yarp-matlab-bindings WITH THE DEFAULT CONFIGURATION (Debug) ===

Check dependencies

Write auxiliary files
write-file /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/yarp-matlab-bindings.build/Debug/ZERO_CHECK.build/Script-575B3B0A126A413291E60F4E.sh
chmod 0755 /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/yarp-matlab-bindings.build/Debug/ZERO_CHECK.build/Script-575B3B0A126A413291E60F4E.sh

PhaseScriptExecution CMake\ Rules /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/yarp-matlab-bindings.build/Debug/ZERO_CHECK.build/Script-575B3B0A126A413291E60F4E.sh
    cd /Users/nunoguedelha/dev/codyco-superbuild/libraries/yarp-matlab-bindings
    /bin/sh -c /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/yarp-matlab-bindings.build/Debug/ZERO_CHECK.build/Script-575B3B0A126A413291E60F4E.sh
echo ""

make -f /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/CMakeScripts/ReRunCMake.make
make[1]: `/Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/CMakeFiles/cmake.check_cache' is up to date.

=== BUILD TARGET yarpMEX OF PROJECT yarp-matlab-bindings WITH THE DEFAULT CONFIGURATION (Debug) ===

Check dependencies

Write auxiliary files
write-file /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/matlab/yarp-matlab-bindings.build/Debug/yarpMEX.build/Script-C7D6494800DA404189D91789.sh
chmod 0755 /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/matlab/yarp-matlab-bindings.build/Debug/yarpMEX.build/Script-C7D6494800DA404189D91789.sh

PhaseScriptExecution CMake\ Rules /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/matlab/yarp-matlab-bindings.build/Debug/yarpMEX.build/Script-C7D6494800DA404189D91789.sh
    cd /Users/nunoguedelha/dev/codyco-superbuild/libraries/yarp-matlab-bindings
    /bin/sh -c /Users/nunoguedelha/dev/codyco-superbuild/build/libraries/yarp-matlab-bindings/matlab/yarp-matlab-bindings.build/Debug/yarpMEX.build/Script-C7D6494800DA404189D91789.sh
echo "Swig source"
Swig source
/usr/local/Cellar/cmake/3.8.0/bin/cmake -E make_directory /Users/nunoguedelha/dev/codyco-superbuild/libraries/yarp-matlab-bindings/matlab
matlab -outdir /Users/nunoguedelha/dev/codyco-superbuild/libraries/yarp-matlab-bindings/matlab -c++ -I/Users/nunoguedelha/dev/yarp/buildXcode/generated_include -I/Users/nunoguedelha/dev/yarp/src/libYARP_OS/include -I/Users/nunoguedelha/dev/yarp/src/libYARP_sig/include -I/Users/nunoguedelha/dev/yarp/src/libYARP_eigen/include -I/Users/nunoguedelha/dev/yarp/src/libYARP_gsl/include -I/Users/nunoguedelha/dev/yarp/src/libYARP_math/include -I/Users/nunoguedelha/dev/yarp/src/libYARP_dev/include -I/Users/nunoguedelha/dev/yarp/src/libYARP_name/include -I/Users/nunoguedelha/dev/yarp/src/libYARP_manager/include -I/Users/nunoguedelha/dev/yarp/src/libYARP_logger/include -I/Applications/MATLAB_R2016b.app/extern/include -o /Users/nunoguedelha/dev/codyco-superbuild/libraries/yarp-matlab-bindings/matlab/yarpMATLAB_wrap.cxx /Users/nunoguedelha/dev/codyco-superbuild/libraries/yarp-matlab-bindings/yarp.i
Unrecognized MATLAB option "outdir".
Unrecognized MATLAB option "c++".
```

The build was launched in `.../codyco-superbuild/build/libraries/yarp-matlab-bindings` after setting the `YARP_GENERATE_MATLAB` flag to `ON`.
I've started this operation from a full updated system and codyco-superbuild on `devel` branch.

**Comment by nunoguedelha** [comment]
@traversaro

**Comment by nunoguedelha** [comment]
As per @traversaro 's and @francesco-romano 's analysis, cmake wasn't locating and using SWIG for generating the bindings. A `find_package(SWIG)` was missing in the `CMakeLists`.
This was fixed by commit f6b6f909c02ed23facfbfad3284c42215cd5d248.
